File: vin/VinCommon/src/scala/com/vinecom/common/scala/tool/topK/TopK.scala
package com.vinecom.common.scala.tool.topK

/**
 * Created by ndn on 7/20/2015.
 */
trait TopK[A] extends Serializable with Iterable[(A, Double)] {
	var length: Int = 0

	def put(other: TopK[A]): TopK[A] = {
		other.foreach(v => put(v))
		this
	}

	def sorted: Array[(A, Double)]
	def put(v: (A, Double)): TopK[A]
}

File: vin/VinCommon/src/java/com/vinecom/common/tool/topK/TopK.java
package com.vinecom.common.tool.topK;

import java.io.Serializable;

import com.vinecom.common.data.Tuple2;

/**
 * Created by ndn on 7/23/2015.
 */
public interface TopK<A> extends Serializable {
	TopK<A> put(Tuple2<A, Double> e);

	TopK<A> put(A a, double d);

	TopK<A> put(TopK<A> other);

	int size();

	Tuple2<A, Double>[] real();

	Tuple2<A, Double>[] all();
}
